NUnitTestMarsRover/MoveCommandTests.cs

```diff
@@ -0,0 +1,312 @@
+﻿using NUnit.Framework;
+using MarsRover;
+using System.Collections.Generic;
+
+namespace NUnitTestMarsRover
+{
+    class MoveCommandTests
+    {
+        // North Heading Tests
```

Comment:
This comment shouldn't be necessary.  We can also greatly simplify this test class.

This class should be split into 4 test classes, one for each class under test.  Make sure they're marked `public` too!

Then, I would use the `[SetUp]` attribute to remove a lot of the boilerplate code between these tests.

An abridged example:

```cs
public class MoveForwardCommandTests()
{
    private NorthHeading _subject;
    private Grid _grid;
    private IEnumerable<Obstacles> _obstacles;
    private MoveForwardCoammnd _subject;
    // any other variables that are shared between tests should be declared here

    [SetUp]
    public void SetUp()
    {
        _subject = new NorthHeading();
        _grid = new Grid(5, 5);
        _obstacles = new List<Obstacle>();
        _subject = new MoveForwardCommand(...);
        // any other variables that are shared between tests should be initialized here
    }

    [Test]
    public void Execute_GivenFacingNorthFromZeroXZeroYPosition_ReturnsCorrectCoordinate()
    {
        _subject.Execute();

        Assert.Multiple(() =>
        {
            Assert.That(rover.YCoordinate, Is.EqualTo(1));
            Assert.That(rover.XCoordinate, Is.EqualTo(0));
        });
    }

    [Test]
    public void Execute_GivenFacingNorthFromFiveXFiveYPositionOnAFiveByFiveGrid_ReturnsCorrectCoordinate()
    {
        _rover.XCoordinate = 5;
        _rover.YCoordinate = 5;

        _moveForward.Execute();

        Assert.Multiple(() =>
        {
            Assert.That(rover.YCoordinate, Is.EqualTo(0));
            Assert.That(rover.XCoordinate, Is.EqualTo(5));
        });
    }

    // other tests
}
```

The advantage of this is it's so much simpler to write subsequent tests!  It more closely follows the AAA pattern for structuring tests, and it also makes refactoring a lot easier.  Remember when we had to go through an touch every single test case when were creating fixup!?  Now we'd only have to touch one test case.  The other thing I did is I changed the name of the test.  It was named `MoveForward_...` but I changed it to `Execute_...` because we're testing the `Execute` function in this test.

Try and apply the techniques here to your other tests.

Reply:
I split up a bunch of these tests and tried to incorporate the example you gave into them.  It does clean stuff up quite a bit!